lib/manager/components/CurrentDeploymentPanel.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+
+import Icon from '@conveyal/woonerf/components/icon'
+import React, { Component } from 'react'
+import {
+  Button,
+  Panel,
+  Label
+} from 'react-bootstrap'
+
+import * as deploymentActions from '../actions/deployments'
+import { formatTimestamp } from '../../common/util/date-time'
+import { getActiveInstanceCount, getServerForId } from '../util/deployment'
+import DeploymentPreviewButton from './DeploymentPreviewButton'
+import EC2InstanceCard from '../../common/components/EC2InstanceCard'
+
+import type { Deployment, OtpServer, ServerJob } from '../../types'
+
+type Props = {
+  deployJob: ?ServerJob,
+  deployment: Deployment,
+  fetchDeployment: typeof deploymentActions.fetchDeployment,
+  server: ?OtpServer
+}
+
+export default class CurrentDeploymentPanel extends Component<Props> {
+  componentWillReceiveProps (nextProps: Props) {
+    // If status message changes while deployment job is in progress, re-fetch
+    // deployment to update EC2 instance status.
+    if (nextProps.deployJob && this.props.deployJob) {
+      if (nextProps.deployJob.status.message !== this.props.deployJob.status.message) {
+        this._fetchDeploymentEC2Instances()
+      }
+    }
+  }
+
+  _fetchDeploymentEC2Instances = () => this.props.fetchDeployment(this.props.deployment.id)
+
+  _getServer = () => {
+    if (this.props.server) return this.props.server
+    const serverId = this.props.deployJob && this.props.deployJob.serverId
+    return getServerForId(serverId, this.props.deployment.project)
+  }
+
+  _getServerLabel = () => {
+    const server = this._getServer()
+    return (
+      <Label>
+        {server
+          ? server.name
+          : this.props.deployment.deployedTo
+        }
+      </Label>
+    )
+  }
+
+  render () {
+    const { deployJob, deployment } = this.props
+    const server = this._getServer()
+    const ec2Info = server && server.ec2Info
+    return (
+      <Panel header={<h3><Icon type='server' /> Current deployment</h3>}>
+        {deployment.deployedTo
+          ? <div>
+            <ul className='list-unstyled'>
+              <li>
+                Deployed to: {this._getServerLabel()}
+              </li>
+              <li>Using ELB?: {ec2Info ? 'Yes' : 'No (built graph over wire)'}</li>
```

Comment:
Since everything below here is so dependent on using an ELB, I think it may be useful to have 3 distinct types of deployments:

- deploy just the bundle to s3
- build graph over wire deployment
- ELB/EC2 deployment

In doing this, we can also modify the forms such that only the needed fields for each deployment type are shown.

Reply:
These are not mutually exclusive types though. For example, you can build the graph over the wire and deploy to s3. Also, the S3 bucket field is necessary for ELB. At the moment this UI is essentially just limited to application admins (i.e., us), so while I think it is important to improve the UI of this form, I think that deserves a separate conversation at this point.

Reply:
I think it would make things simpler if those 3 items were mutually exclusive. Anyways, this will be something to explore as we get feedback from clients.